executor/executor_pkg_test.go

```diff
@@ -43,7 +43,7 @@ import (
 	"github.com/pingcap/tidb/util/tableutil"
 )
 
-var _ = Suite(&testExecSuite{})
+var _ = SerialSuites(&testExecSuite{})
```

Comment:
testExecSuite this test call `RegisterVirtualTable` which modify a global table driver struct and some others are visiting it, then DATA RACE.

Reply:
Where does it call `RegisterVirtualTable`? I didn't find it...

Reply:
```
==================
WARNING: DATA RACE
Write at 0x00c000651a88 by goroutine 60:
  github.com/pingcap/tidb/sessionctx/variable.(*SessionVars).InitStatementContext()
      /home/genius/project/src/github.com/pingcap/tidb/sessionctx/variable/session.go:885 +0x3124
  github.com/pingcap/tidb/executor.ResetContextOfStmt()
      /home/genius/project/src/github.com/pingcap/tidb/executor/executor.go:1662 +0x3180
  github.com/pingcap/tidb/session.(*session).ExecuteStmt()
      /home/genius/project/src/github.com/pingcap/tidb/session/session.go:1606 +0x204
  github.com/pingcap/tidb/util/testkit.(*TestKit).Exec()
      /home/genius/project/src/github.com/pingcap/tidb/util/testkit/testkit.go:170 +0x2fc
  github.com/pingcap/tidb/util/testkit.(*TestKit).MustQuery()
      /home/genius/project/src/github.com/pingcap/tidb/util/testkit/testkit.go:335 +0x219
  github.com/pingcap/tidb/executor_test.(*testSuite).TestSummaryFailedUpdate()
      /home/genius/project/src/github.com/pingcap/tidb/executor/executor_test.go:5640 +0x71c
  github.com/pingcap/tidb/util/memory.(*Tracker).Consume()
      /home/genius/project/src/github.com/pingcap/tidb/util/memory/tracker.go:343 +0x21a
  github.com/pingcap/tidb/executor.(*HashAggExec).initForParallelExec()
      /home/genius/project/src/github.com/pingcap/tidb/executor/aggregate.go:384 +0xcd2
  github.com/pingcap/tidb/executor.(*HashAggExec).Open()
      /home/genius/project/src/github.com/pingcap/tidb/executor/aggregate.go:320 +0x3e6
  github.com/pingcap/tidb/executor.(*baseExecutor).Open()
      /home/genius/project/src/github.com/pingcap/tidb/executor/executor.go:168 +0xae
  github.com/pingcap/tidb/executor.(*HashJoinExec).Open()
      /home/genius/project/src/github.com/pingcap/tidb/executor/join.go:157 +0x71
  github.com/pingcap/tidb/executor.(*UpdateExec).Open()
      /home/genius/project/src/github.com/pingcap/tidb/executor/update.go:428 +0x306
  github.com/pingcap/tidb/executor.(*ExecStmt).Exec()
      /home/genius/project/src/github.com/pingcap/tidb/executor/adapter.go:388 +0x4e2
  github.com/pingcap/tidb/session.runStmt()
      /home/genius/project/src/github.com/pingcap/tidb/session/session.go:1755 +0x323
  github.com/pingcap/tidb/session.(*session).ExecuteStmt()
      /home/genius/project/src/github.com/pingcap/tidb/session/session.go:1650 +0xcd7
  github.com/pingcap/tidb/util/testkit.(*TestKit).Exec()
      /home/genius/project/src/github.com/pingcap/tidb/util/testkit/testkit.go:170 +0x2fc
  github.com/pingcap/tidb/util/testkit.(*TestKit).ExecToErr()
      /home/genius/project/src/github.com/pingcap/tidb/util/testkit/testkit.go:367 +0x96
  github.com/pingcap/tidb/executor_test.(*testSuite).TestSummaryFailedUpdate()
      /home/genius/project/src/github.com/pingcap/tidb/executor/executor_test.go:5636 +0x544
  runtime.call16()
      /snap/go/7954/src/runtime/asm_amd64.s:550 +0x3d
  reflect.Value.Call()
      /snap/go/7954/src/reflect/value.go:337 +0xd8
  github.com/pingcap/check.(*suiteRunner).forkTest.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:850 +0xb3b
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:739 +0x11d

Previous write at 0x00c000651a88 by goroutine 69:
  sync/atomic.CompareAndSwapInt64()
      /snap/go/7954/src/runtime/race_amd64.s:327 +0xb
  github.com/pingcap/tidb/util/memory.(*Tracker).Consume()
      /home/genius/project/src/github.com/pingcap/tidb/util/memory/tracker.go:324 +0x171
  github.com/pingcap/tidb/store/copr.(*copIteratorWorker).sendToRespCh()
      /home/genius/project/src/github.com/pingcap/tidb/store/copr/coprocessor.go:517 +0x290
  github.com/pingcap/tidb/store/copr.(*copIteratorWorker).handleCopResponse()
      /home/genius/project/src/github.com/pingcap/tidb/store/copr/coprocessor.go:979 +0xc39
  github.com/pingcap/tidb/store/copr.(*copIteratorWorker).handleTaskOnce()
      /home/genius/project/src/github.com/pingcap/tidb/store/copr/coprocessor.go:748 +0x1264
  github.com/pingcap/tidb/store/copr.(*copIteratorWorker).handleTask()
      /home/genius/project/src/github.com/pingcap/tidb/store/copr/coprocessor.go:640 +0x1e4
  github.com/pingcap/tidb/store/copr.(*copIteratorWorker).run()
      /home/genius/project/src/github.com/pingcap/tidb/store/copr/coprocessor.go:377 +0x15a

Goroutine 60 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).forkTest()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:832 +0x1c4
  github.com/pingcap/check.(*suiteRunner).doRun()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:666 +0x144
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:650 +0x105

Goroutine 69 (finished) created at:
  github.com/pingcap/tidb/store/copr.(*copIterator).open()
      /home/genius/project/src/github.com/pingcap/tidb/store/copr/coprocessor.go:414 +0xe7
  github.com/pingcap/tidb/store/copr.(*CopClient).Send()
      /home/genius/project/src/github.com/pingcap/tidb/store/copr/coprocessor.go:127 +0xc49
  github.com/pingcap/tidb/distsql.Select()
      /home/genius/project/src/github.com/pingcap/tidb/distsql/distsql.go:77 +0x328
  github.com/pingcap/tidb/distsql.SelectWithRuntimeStats()
      /home/genius/project/src/github.com/pingcap/tidb/distsql/distsql.go:126 +0xbd
  github.com/pingcap/tidb/executor.selectResultHook.SelectResult()
      /home/genius/project/src/github.com/pingcap/tidb/executor/table_reader.go:51 +0x22b
  github.com/pingcap/tidb/executor.(*TableReaderExecutor).buildResp()
      /home/genius/project/src/github.com/pingcap/tidb/executor/table_reader.go:303 +0xa15
  github.com/pingcap/tidb/executor.(*TableReaderExecutor).Open()
      /home/genius/project/src/github.com/pingcap/tidb/executor/table_reader.go:197 +0xba6
  github.com/pingcap/tidb/executor.(*baseExecutor).Open()
      /home/genius/project/src/github.com/pingcap/tidb/executor/executor.go:168 +0xae
  github.com/pingcap/tidb/executor.(*HashAggExec).Open()
      /home/genius/project/src/github.com/pingcap/tidb/executor/aggregate.go:301 +0x71
  github.com/pingcap/tidb/executor.(*baseExecutor).Open()
      /home/genius/project/src/github.com/pingcap/tidb/executor/executor.go:168 +0xae
  github.com/pingcap/tidb/executor.(*HashJoinExec).Open()
      /home/genius/project/src/github.com/pingcap/tidb/executor/join.go:157 +0x71
  github.com/pingcap/tidb/executor.(*UpdateExec).Open()
      /home/genius/project/src/github.com/pingcap/tidb/executor/update.go:428 +0x306
  github.com/pingcap/tidb/executor.(*ExecStmt).Exec()
      /home/genius/project/src/github.com/pingcap/tidb/executor/adapter.go:388 +0x4e2
  github.com/pingcap/tidb/session.runStmt()
      /home/genius/project/src/github.com/pingcap/tidb/session/session.go:1755 +0x323
  github.com/pingcap/tidb/session.(*session).ExecuteStmt()
      /home/genius/project/src/github.com/pingcap/tidb/session/session.go:1650 +0xcd7
  github.com/pingcap/tidb/util/testkit.(*TestKit).Exec()
      /home/genius/project/src/github.com/pingcap/tidb/util/testkit/testkit.go:170 +0x2fc
  github.com/pingcap/tidb/util/testkit.(*TestKit).ExecToErr()
      /home/genius/project/src/github.com/pingcap/tidb/util/testkit/testkit.go:367 +0x96
  github.com/pingcap/tidb/executor_test.(*testSuite).TestSummaryFailedUpdate()
      /home/genius/project/src/github.com/pingcap/tidb/executor/executor_test.go:5636 +0x544
  runtime.call16()
      /snap/go/7954/src/runtime/asm_amd64.s:550 +0x3d
  reflect.Value.Call()
      /snap/go/7954/src/reflect/value.go:337 +0xd8
  github.com/pingcap/check.(*suiteRunner).forkTest.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:850 +0xb3b
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:739 +0x11d
```

Reply:
```
==================
WARNING: DATA RACE
Write at 0x0000081a12c0 by goroutine 96:
  github.com/pingcap/tidb/infoschema.RegisterVirtualTable()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:582 +0x3dd
  github.com/pingcap/tidb/infoschema/perfschema.Init.func1()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/perfschema/init.go:71 +0x2c9
  sync.(*Once).doSlow()
      /snap/go/7954/src/sync/once.go:68 +0x109
  sync.(*Once).Do()
      /snap/go/7954/src/sync/once.go:59 +0x68
  github.com/pingcap/tidb/infoschema/perfschema.Init()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/perfschema/init.go:74 +0x6f
  github.com/pingcap/tidb/domain.(*Domain).Init()
      /home/genius/project/src/github.com/pingcap/tidb/domain/domain.go:656 +0x49
  github.com/pingcap/tidb/session.(*domainMap).Get.func1()
      /home/genius/project/src/github.com/pingcap/tidb/session/tidb.go:78 +0x6dd
  github.com/pingcap/tidb/util.RunWithRetry()
      /home/genius/project/src/github.com/pingcap/tidb/util/misc.go:64 +0x97
  github.com/pingcap/tidb/session.(*domainMap).Get()
      /home/genius/project/src/github.com/pingcap/tidb/session/tidb.go:69 +0x2bd
  github.com/pingcap/tidb/session.createSessionWithOpt()
      /home/genius/project/src/github.com/pingcap/tidb/session/session.go:2665 +0x7e
  github.com/pingcap/tidb/session.createSession()
      /home/genius/project/src/github.com/pingcap/tidb/session/session.go:2661 +0x6c
  github.com/pingcap/tidb/session.runInBootstrapSession()
      /home/genius/project/src/github.com/pingcap/tidb/session/session.go:2644 +0x45
  github.com/pingcap/tidb/session.BootstrapSession()
      /home/genius/project/src/github.com/pingcap/tidb/session/session.go:2507 +0xfd
  github.com/pingcap/tidb/executor_test.(*baseTestSuite).SetUpSuite()
      /home/genius/project/src/github.com/pingcap/tidb/executor/executor_test.go:201 +0x17a
  github.com/pingcap/tidb/executor_test.(*testSuite).SetUpSuite()
      <autogenerated>:1 +0x64
  runtime.call16()
      /snap/go/7954/src/runtime/asm_amd64.s:550 +0x3d
  reflect.Value.Call()
      /snap/go/7954/src/reflect/value.go:337 +0xd8
  github.com/pingcap/check.(*suiteRunner).runFixture.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:799 +0x1d0
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:739 +0x11d

Previous read at 0x0000081a12c0 by goroutine 228:
  github.com/pingcap/tidb/infoschema.(*Builder).InitWithDBInfos()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:536 +0x1da
  github.com/pingcap/tidb/executor.newSlowQueryRetriever()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:54 +0x213
  github.com/pingcap/tidb/executor.parseSlowLog()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:67 +0x49
  github.com/pingcap/tidb/executor.(*testExecSuite).TestParseSlowLogFile()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:143 +0x292
  runtime.call16()
      /snap/go/7954/src/runtime/asm_amd64.s:550 +0x3d
  reflect.Value.Call()
      /snap/go/7954/src/reflect/value.go:337 +0xd8
  github.com/pingcap/check.(*suiteRunner).forkTest.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:850 +0xb3b
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:739 +0x11d

Goroutine 96 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).runFunc()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:751 +0x84
  github.com/pingcap/check.(*suiteRunner).runFixture()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:795 +0x3d
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:647 +0x86

Goroutine 228 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).forkTest()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:832 +0x1c4
  github.com/pingcap/check.(*suiteRunner).doRun()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:666 +0x144
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:650 +0x105
==================
==================
WARNING: DATA RACE
Read at 0x00c000bfad50 by goroutine 228:
  github.com/pingcap/tidb/infoschema.(*Builder).InitWithDBInfos()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:536 +0x224
  github.com/pingcap/tidb/executor.newSlowQueryRetriever()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:54 +0x213
  github.com/pingcap/tidb/executor.parseSlowLog()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:67 +0x49
  github.com/pingcap/tidb/executor.(*testExecSuite).TestParseSlowLogFile()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:166 +0x7ea
  runtime.call16()
      /snap/go/7954/src/runtime/asm_amd64.s:550 +0x3d
  reflect.Value.Call()
      /snap/go/7954/src/reflect/value.go:337 +0xd8
  github.com/pingcap/check.(*suiteRunner).forkTest.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:850 +0xb3b
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:739 +0x11d

Previous write at 0x00c000bfad50 by goroutine 96:
  [failed to restore the stack]

Goroutine 228 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).forkTest()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:832 +0x1c4
  github.com/pingcap/check.(*suiteRunner).doRun()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:666 +0x144
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:650 +0x105

Goroutine 96 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).runFunc()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:751 +0x84
  github.com/pingcap/check.(*suiteRunner).runFixture()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:795 +0x3d
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:647 +0x86
==================
==================
WARNING: DATA RACE
Read at 0x00c000e12420 by goroutine 228:
  github.com/pingcap/tidb/infoschema.(*Builder).InitWithDBInfos()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:537 +0x244
  github.com/pingcap/tidb/executor.newSlowQueryRetriever()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:54 +0x213
  github.com/pingcap/tidb/executor.parseSlowLog()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:67 +0x49
  github.com/pingcap/tidb/executor.(*testExecSuite).TestParseSlowLogFile()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:166 +0x7ea
  runtime.call16()
      /snap/go/7954/src/runtime/asm_amd64.s:550 +0x3d
  reflect.Value.Call()
      /snap/go/7954/src/reflect/value.go:337 +0xd8
  github.com/pingcap/check.(*suiteRunner).forkTest.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:850 +0xb3b
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:739 +0x11d

Previous write at 0x00c000e12420 by goroutine 96:
  [failed to restore the stack]

Goroutine 228 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).forkTest()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:832 +0x1c4
  github.com/pingcap/check.(*suiteRunner).doRun()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:666 +0x144
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:650 +0x105

Goroutine 96 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).runFunc()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:751 +0x84
  github.com/pingcap/check.(*suiteRunner).runFixture()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:795 +0x3d
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:647 +0x86
==================
==================
WARNING: DATA RACE
Read at 0x00c000e12428 by goroutine 228:
  github.com/pingcap/tidb/infoschema.(*Builder).InitWithDBInfos()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:537 +0x25e
  github.com/pingcap/tidb/executor.newSlowQueryRetriever()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:54 +0x213
  github.com/pingcap/tidb/executor.parseSlowLog()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:67 +0x49
  github.com/pingcap/tidb/executor.(*testExecSuite).TestParseSlowLogFile()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:166 +0x7ea
  runtime.call16()
      /snap/go/7954/src/runtime/asm_amd64.s:550 +0x3d
  reflect.Value.Call()
      /snap/go/7954/src/reflect/value.go:337 +0xd8
  github.com/pingcap/check.(*suiteRunner).forkTest.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:850 +0xb3b
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:739 +0x11d

Previous write at 0x00c000e12428 by goroutine 96:
  [failed to restore the stack]

Goroutine 228 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).forkTest()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:832 +0x1c4
  github.com/pingcap/check.(*suiteRunner).doRun()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:666 +0x144
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:650 +0x105

Goroutine 96 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).runFunc()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:751 +0x84
  github.com/pingcap/check.(*suiteRunner).runFixture()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:795 +0x3d
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:647 +0x86
==================
==================
WARNING: DATA RACE
Read at 0x00c000b04048 by goroutine 228:
  github.com/pingcap/tidb/infoschema.(*Builder).createSchemaTablesForDB()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:555 +0x64
  github.com/pingcap/tidb/infoschema.(*Builder).InitWithDBInfos()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:537 +0x287
  github.com/pingcap/tidb/executor.newSlowQueryRetriever()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:54 +0x213
  github.com/pingcap/tidb/executor.parseSlowLog()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:67 +0x49
  github.com/pingcap/tidb/executor.(*testExecSuite).TestParseSlowLogFile()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:166 +0x7ea
  runtime.call16()
      /snap/go/7954/src/runtime/asm_amd64.s:550 +0x3d
  reflect.Value.Call()
      /snap/go/7954/src/reflect/value.go:337 +0xd8
  github.com/pingcap/check.(*suiteRunner).forkTest.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:850 +0xb3b
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:739 +0x11d

Previous write at 0x00c000b04048 by goroutine 96:
  [failed to restore the stack]

Goroutine 228 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).forkTest()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:832 +0x1c4
  github.com/pingcap/check.(*suiteRunner).doRun()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:666 +0x144
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:650 +0x105

Goroutine 96 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).runFunc()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:751 +0x84
  github.com/pingcap/check.(*suiteRunner).runFixture()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:795 +0x3d
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:647 +0x86
==================
==================
WARNING: DATA RACE
Read at 0x00c000b04018 by goroutine 228:
  github.com/pingcap/tidb/infoschema.(*Builder).createSchemaTablesForDB()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:557 +0x178
  github.com/pingcap/tidb/infoschema.(*Builder).InitWithDBInfos()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:537 +0x287
  github.com/pingcap/tidb/executor.newSlowQueryRetriever()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:54 +0x213
  github.com/pingcap/tidb/executor.parseSlowLog()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:67 +0x49
  github.com/pingcap/tidb/executor.(*testExecSuite).TestParseSlowLogFile()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:166 +0x7ea
  runtime.call16()
      /snap/go/7954/src/runtime/asm_amd64.s:550 +0x3d
  reflect.Value.Call()
      /snap/go/7954/src/reflect/value.go:337 +0xd8
  github.com/pingcap/check.(*suiteRunner).forkTest.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:850 +0xb3b
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:739 +0x11d

Previous write at 0x00c000b04018 by goroutine 96:
  [failed to restore the stack]

Goroutine 228 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).forkTest()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:832 +0x1c4
  github.com/pingcap/check.(*suiteRunner).doRun()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:666 +0x144
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:650 +0x105

Goroutine 96 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).runFunc()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:751 +0x84
  github.com/pingcap/check.(*suiteRunner).runFixture()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:795 +0x3d
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:647 +0x86
==================
==================
WARNING: DATA RACE
Read at 0x00c000b04000 by goroutine 228:
  github.com/pingcap/tidb/infoschema.(*Builder).createSchemaTablesForDB()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:560 +0x2bd
  github.com/pingcap/tidb/infoschema.(*Builder).InitWithDBInfos()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:537 +0x287
  github.com/pingcap/tidb/executor.newSlowQueryRetriever()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:54 +0x213
  github.com/pingcap/tidb/executor.parseSlowLog()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:67 +0x49
  github.com/pingcap/tidb/executor.(*testExecSuite).TestParseSlowLogFile()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:166 +0x7ea
  runtime.call16()
      /snap/go/7954/src/runtime/asm_amd64.s:550 +0x3d
  reflect.Value.Call()
      /snap/go/7954/src/reflect/value.go:337 +0xd8
  github.com/pingcap/check.(*suiteRunner).forkTest.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:850 +0xb3b
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:739 +0x11d

Previous write at 0x00c000b04000 by goroutine 96:
  [failed to restore the stack]

Goroutine 228 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).forkTest()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:832 +0x1c4
  github.com/pingcap/check.(*suiteRunner).doRun()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:666 +0x144
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:650 +0x105

Goroutine 96 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).runFunc()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:751 +0x84
  github.com/pingcap/check.(*suiteRunner).runFixture()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:795 +0x3d
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:647 +0x86
==================
==================
WARNING: DATA RACE
Read at 0x00c000901bf8 by goroutine 228:
  github.com/pingcap/parser/model.(*TableInfo).GetDBID()
      /home/genius/go/pkg/mod/github.com/pingcap/parser@v0.0.0-20210802034743-dd9b189324ce/model/model.go:485 +0x66
  github.com/pingcap/tidb/meta/autoid.NewAllocatorsFromTblInfo()
      /home/genius/project/src/github.com/pingcap/tidb/meta/autoid/autoid.go:494 +0x7d
  github.com/pingcap/tidb/infoschema.(*Builder).createSchemaTablesForDB()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:560 +0x2f5
  github.com/pingcap/tidb/infoschema.(*Builder).InitWithDBInfos()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:537 +0x287
  github.com/pingcap/tidb/executor.newSlowQueryRetriever()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:54 +0x213
  github.com/pingcap/tidb/executor.parseSlowLog()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:67 +0x49
  github.com/pingcap/tidb/executor.(*testExecSuite).TestParseSlowLogFile()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:166 +0x7ea
  runtime.call16()
      /snap/go/7954/src/runtime/asm_amd64.s:550 +0x3d
  reflect.Value.Call()
      /snap/go/7954/src/reflect/value.go:337 +0xd8
  github.com/pingcap/check.(*suiteRunner).forkTest.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:850 +0xb3b
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:739 +0x11d

Previous write at 0x00c000901bf8 by goroutine 96:
  [failed to restore the stack]

Goroutine 228 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).forkTest()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:832 +0x1c4
  github.com/pingcap/check.(*suiteRunner).doRun()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:666 +0x144
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:650 +0x105

Goroutine 96 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).runFunc()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:751 +0x84
  github.com/pingcap/check.(*suiteRunner).runFixture()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:795 +0x3d
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:647 +0x86
==================
==================
WARNING: DATA RACE
Read at 0x00c000901bc8 by goroutine 228:
  github.com/pingcap/tidb/meta/autoid.NewAllocatorsFromTblInfo()
      /home/genius/project/src/github.com/pingcap/tidb/meta/autoid/autoid.go:495 +0x99
  github.com/pingcap/tidb/infoschema.(*Builder).createSchemaTablesForDB()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:560 +0x2f5
  github.com/pingcap/tidb/infoschema.(*Builder).InitWithDBInfos()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:537 +0x287
  github.com/pingcap/tidb/executor.newSlowQueryRetriever()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:54 +0x213
  github.com/pingcap/tidb/executor.parseSlowLog()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:67 +0x49
  github.com/pingcap/tidb/executor.(*testExecSuite).TestParseSlowLogFile()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:166 +0x7ea
  runtime.call16()
      /snap/go/7954/src/runtime/asm_amd64.s:550 +0x3d
  reflect.Value.Call()
      /snap/go/7954/src/reflect/value.go:337 +0xd8
  github.com/pingcap/check.(*suiteRunner).forkTest.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:850 +0xb3b
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:739 +0x11d

Previous write at 0x00c000901bc8 by goroutine 96:
  [failed to restore the stack]

Goroutine 228 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).forkTest()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:832 +0x1c4
  github.com/pingcap/check.(*suiteRunner).doRun()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:666 +0x144
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:650 +0x105

Goroutine 96 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).runFunc()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:751 +0x84
  github.com/pingcap/check.(*suiteRunner).runFixture()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:795 +0x3d
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:647 +0x86
==================
==================
WARNING: DATA RACE
Read at 0x00c000901ba9 by goroutine 228:
  github.com/pingcap/tidb/meta/autoid.NewAllocatorsFromTblInfo()
      /home/genius/project/src/github.com/pingcap/tidb/meta/autoid/autoid.go:497 +0xbd
  github.com/pingcap/tidb/infoschema.(*Builder).createSchemaTablesForDB()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:560 +0x2f5
  github.com/pingcap/tidb/infoschema.(*Builder).InitWithDBInfos()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:537 +0x287
  github.com/pingcap/tidb/executor.newSlowQueryRetriever()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:54 +0x213
  github.com/pingcap/tidb/executor.parseSlowLog()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:67 +0x49
  github.com/pingcap/tidb/executor.(*testExecSuite).TestParseSlowLogFile()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:166 +0x7ea
  runtime.call16()
      /snap/go/7954/src/runtime/asm_amd64.s:550 +0x3d
  reflect.Value.Call()
      /snap/go/7954/src/reflect/value.go:337 +0xd8
  github.com/pingcap/check.(*suiteRunner).forkTest.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:850 +0xb3b
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:739 +0x11d

Previous write at 0x00c000901ba8 by goroutine 96:
  [failed to restore the stack]

Goroutine 228 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).forkTest()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:832 +0x1c4
  github.com/pingcap/check.(*suiteRunner).doRun()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:666 +0x144
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:650 +0x105

Goroutine 96 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).runFunc()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:751 +0x84
  github.com/pingcap/check.(*suiteRunner).runFixture()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:795 +0x3d
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:647 +0x86
==================
==================
WARNING: DATA RACE
Read at 0x00c000901b48 by goroutine 228:
  github.com/pingcap/parser/model.(*TableInfo).GetAutoIncrementColInfo()
      /home/genius/go/pkg/mod/github.com/pingcap/parser@v0.0.0-20210802034743-dd9b189324ce/model/model.go:535 +0x44
  github.com/pingcap/tidb/meta/autoid.NewAllocatorsFromTblInfo()
      /home/genius/project/src/github.com/pingcap/tidb/meta/autoid/autoid.go:498 +0xe4
  github.com/pingcap/tidb/infoschema.(*Builder).createSchemaTablesForDB()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:560 +0x2f5
  github.com/pingcap/tidb/infoschema.(*Builder).InitWithDBInfos()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:537 +0x287
  github.com/pingcap/tidb/executor.newSlowQueryRetriever()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:54 +0x213
  github.com/pingcap/tidb/executor.parseSlowLog()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:67 +0x49
  github.com/pingcap/tidb/executor.(*testExecSuite).TestParseSlowLogFile()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:166 +0x7ea
  runtime.call16()
      /snap/go/7954/src/runtime/asm_amd64.s:550 +0x3d
  reflect.Value.Call()
      /snap/go/7954/src/reflect/value.go:337 +0xd8
  github.com/pingcap/check.(*suiteRunner).forkTest.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:850 +0xb3b
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:739 +0x11d

Previous write at 0x00c000901b48 by goroutine 96:
  [failed to restore the stack]

Goroutine 228 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).forkTest()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:832 +0x1c4
  github.com/pingcap/check.(*suiteRunner).doRun()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:666 +0x144
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:650 +0x105

Goroutine 96 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).runFunc()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:751 +0x84
  github.com/pingcap/check.(*suiteRunner).runFixture()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:795 +0x3d
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:647 +0x86
==================
==================
WARNING: DATA RACE
Read at 0x00c00cdb26f0 by goroutine 228:
  github.com/pingcap/parser/model.(*TableInfo).GetAutoIncrementColInfo()
      /home/genius/go/pkg/mod/github.com/pingcap/parser@v0.0.0-20210802034743-dd9b189324ce/model/model.go:536 +0xb0
  github.com/pingcap/tidb/meta/autoid.NewAllocatorsFromTblInfo()
      /home/genius/project/src/github.com/pingcap/tidb/meta/autoid/autoid.go:498 +0xe4
  github.com/pingcap/tidb/infoschema.(*Builder).createSchemaTablesForDB()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:560 +0x2f5
  github.com/pingcap/tidb/infoschema.(*Builder).InitWithDBInfos()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:537 +0x287
  github.com/pingcap/tidb/executor.newSlowQueryRetriever()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:54 +0x213
  github.com/pingcap/tidb/executor.parseSlowLog()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:67 +0x49
  github.com/pingcap/tidb/executor.(*testExecSuite).TestParseSlowLogFile()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:166 +0x7ea
  runtime.call16()
      /snap/go/7954/src/runtime/asm_amd64.s:550 +0x3d
  reflect.Value.Call()
      /snap/go/7954/src/reflect/value.go:337 +0xd8
  github.com/pingcap/check.(*suiteRunner).forkTest.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:850 +0xb3b
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:739 +0x11d

Previous write at 0x00c00cdb26f0 by goroutine 96:
  [failed to restore the stack]

Goroutine 228 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).forkTest()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:832 +0x1c4
  github.com/pingcap/check.(*suiteRunner).doRun()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:666 +0x144
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:650 +0x105

Goroutine 96 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).runFunc()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:751 +0x84
  github.com/pingcap/check.(*suiteRunner).runFixture()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:795 +0x3d
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:647 +0x86
==================
==================
WARNING: DATA RACE
Read at 0x00c001056680 by goroutine 228:
  github.com/pingcap/parser/model.(*TableInfo).GetAutoIncrementColInfo()
      /home/genius/go/pkg/mod/github.com/pingcap/parser@v0.0.0-20210802034743-dd9b189324ce/model/model.go:535 +0x8b
  github.com/pingcap/tidb/meta/autoid.NewAllocatorsFromTblInfo()
      /home/genius/project/src/github.com/pingcap/tidb/meta/autoid/autoid.go:498 +0xe4
  github.com/pingcap/tidb/infoschema.(*Builder).createSchemaTablesForDB()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:560 +0x2f5
  github.com/pingcap/tidb/infoschema.(*Builder).InitWithDBInfos()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:537 +0x287
  github.com/pingcap/tidb/executor.newSlowQueryRetriever()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:54 +0x213
  github.com/pingcap/tidb/executor.parseSlowLog()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:67 +0x49
  github.com/pingcap/tidb/executor.(*testExecSuite).TestParseSlowLogFile()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:166 +0x7ea
  runtime.call16()
      /snap/go/7954/src/runtime/asm_amd64.s:550 +0x3d
  reflect.Value.Call()
      /snap/go/7954/src/reflect/value.go:337 +0xd8
  github.com/pingcap/check.(*suiteRunner).forkTest.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:850 +0xb3b
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:739 +0x11d

Previous write at 0x00c001056680 by goroutine 96:
  [failed to restore the stack]

Goroutine 228 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).forkTest()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:832 +0x1c4
  github.com/pingcap/check.(*suiteRunner).doRun()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:666 +0x144
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:650 +0x105

Goroutine 96 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).runFunc()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:751 +0x84
  github.com/pingcap/check.(*suiteRunner).runFixture()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:795 +0x3d
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:647 +0x86
==================
==================
WARNING: DATA RACE
Read at 0x00c000901c10 by goroutine 228:
  github.com/pingcap/parser/model.(*TableInfo).ContainsAutoRandomBits()
      /home/genius/go/pkg/mod/github.com/pingcap/parser@v0.0.0-20210802034743-dd9b189324ce/model/model.go:553 +0x29d
  github.com/pingcap/tidb/meta/autoid.NewAllocatorsFromTblInfo()
      /home/genius/project/src/github.com/pingcap/tidb/meta/autoid/autoid.go:503 +0x285
  github.com/pingcap/tidb/infoschema.(*Builder).createSchemaTablesForDB()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:560 +0x2f5
  github.com/pingcap/tidb/infoschema.(*Builder).InitWithDBInfos()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:537 +0x287
  github.com/pingcap/tidb/executor.newSlowQueryRetriever()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:54 +0x213
  github.com/pingcap/tidb/executor.parseSlowLog()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:67 +0x49
  github.com/pingcap/tidb/executor.(*testExecSuite).TestParseSlowLogFile()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:166 +0x7ea
  runtime.call16()
      /snap/go/7954/src/runtime/asm_amd64.s:550 +0x3d
  reflect.Value.Call()
      /snap/go/7954/src/reflect/value.go:337 +0xd8
  github.com/pingcap/check.(*suiteRunner).forkTest.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:850 +0xb3b
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:739 +0x11d

Previous write at 0x00c000901c10 by goroutine 96:
  [failed to restore the stack]

Goroutine 228 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).forkTest()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:832 +0x1c4
  github.com/pingcap/check.(*suiteRunner).doRun()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:666 +0x144
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:650 +0x105

Goroutine 96 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).runFunc()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:751 +0x84
  github.com/pingcap/check.(*suiteRunner).runFixture()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:795 +0x3d
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:647 +0x86
==================
==================
WARNING: DATA RACE
Read at 0x00c000901c40 by goroutine 228:
  github.com/pingcap/parser/model.(*TableInfo).IsSequence()
      /home/genius/go/pkg/mod/github.com/pingcap/parser@v0.0.0-20210802034743-dd9b189324ce/model/model.go:637 +0x2ef
  github.com/pingcap/tidb/meta/autoid.NewAllocatorsFromTblInfo()
      /home/genius/project/src/github.com/pingcap/tidb/meta/autoid/autoid.go:508 +0x300
  github.com/pingcap/tidb/infoschema.(*Builder).createSchemaTablesForDB()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:560 +0x2f5
  github.com/pingcap/tidb/infoschema.(*Builder).InitWithDBInfos()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:537 +0x287
  github.com/pingcap/tidb/executor.newSlowQueryRetriever()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:54 +0x213
  github.com/pingcap/tidb/executor.parseSlowLog()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:67 +0x49
  github.com/pingcap/tidb/executor.(*testExecSuite).TestParseSlowLogFile()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:166 +0x7ea
  runtime.call16()
      /snap/go/7954/src/runtime/asm_amd64.s:550 +0x3d
  reflect.Value.Call()
      /snap/go/7954/src/reflect/value.go:337 +0xd8
  github.com/pingcap/check.(*suiteRunner).forkTest.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:850 +0xb3b
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:739 +0x11d

Previous write at 0x00c000901c40 by goroutine 96:
  [failed to restore the stack]

Goroutine 228 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).forkTest()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:832 +0x1c4
  github.com/pingcap/check.(*suiteRunner).doRun()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:666 +0x144
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:650 +0x105

Goroutine 96 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).runFunc()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:751 +0x84
  github.com/pingcap/check.(*suiteRunner).runFixture()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:795 +0x3d
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:647 +0x86
==================
==================
WARNING: DATA RACE
Read at 0x00c000901b18 by goroutine 228:
  github.com/pingcap/tidb/infoschema/perfschema.tableFromMeta()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/perfschema/tables.go:130 +0x77
  github.com/pingcap/tidb/infoschema.(*Builder).createSchemaTablesForDB()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:562 +0x32c
  github.com/pingcap/tidb/infoschema.(*Builder).InitWithDBInfos()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:537 +0x287
  github.com/pingcap/tidb/executor.newSlowQueryRetriever()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:54 +0x213
  github.com/pingcap/tidb/executor.parseSlowLog()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:67 +0x49
  github.com/pingcap/tidb/executor.(*testExecSuite).TestParseSlowLogFile()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:166 +0x7ea
  runtime.call16()
      /snap/go/7954/src/runtime/asm_amd64.s:550 +0x3d
  reflect.Value.Call()
      /snap/go/7954/src/reflect/value.go:337 +0xd8
  github.com/pingcap/check.(*suiteRunner).forkTest.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:850 +0xb3b
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:739 +0x11d

Previous write at 0x00c000901b18 by goroutine 96:
  [failed to restore the stack]

Goroutine 228 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).forkTest()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:832 +0x1c4
  github.com/pingcap/check.(*suiteRunner).doRun()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:666 +0x144
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:650 +0x105

Goroutine 96 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).runFunc()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:751 +0x84
  github.com/pingcap/check.(*suiteRunner).runFixture()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:795 +0x3d
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:647 +0x86
==================
==================
WARNING: DATA RACE
Read at 0x00c000901b60 by goroutine 228:
  github.com/pingcap/tidb/infoschema/perfschema.initTableIndices()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/perfschema/tables.go:209 +0x76
  github.com/pingcap/tidb/infoschema/perfschema.createPerfSchemaTable()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/perfschema/tables.go:150 +0x405
  github.com/pingcap/tidb/infoschema/perfschema.tableFromMeta()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/perfschema/tables.go:134 +0x199
  github.com/pingcap/tidb/infoschema.(*Builder).createSchemaTablesForDB()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:562 +0x32c
  github.com/pingcap/tidb/infoschema.(*Builder).InitWithDBInfos()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:537 +0x287
  github.com/pingcap/tidb/executor.newSlowQueryRetriever()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:54 +0x213
  github.com/pingcap/tidb/executor.parseSlowLog()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:67 +0x49
  github.com/pingcap/tidb/executor.(*testExecSuite).TestParseSlowLogFile()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:166 +0x7ea
  runtime.call16()
      /snap/go/7954/src/runtime/asm_amd64.s:550 +0x3d
  reflect.Value.Call()
      /snap/go/7954/src/reflect/value.go:337 +0xd8
  github.com/pingcap/check.(*suiteRunner).forkTest.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:850 +0xb3b
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:739 +0x11d

Previous write at 0x00c000901b60 by goroutine 96:
  [failed to restore the stack]

Goroutine 228 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).forkTest()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:832 +0x1c4
  github.com/pingcap/check.(*suiteRunner).doRun()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:666 +0x144
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:650 +0x105

Goroutine 96 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).runFunc()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:751 +0x84
  github.com/pingcap/check.(*suiteRunner).runFixture()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:795 +0x3d
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:647 +0x86
==================
==================
WARNING: DATA RACE
Read at 0x00c000901b00 by goroutine 228:
  github.com/pingcap/tidb/infoschema.(*Builder).createSchemaTablesForDB()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:567 +0x424
  github.com/pingcap/tidb/infoschema.(*Builder).InitWithDBInfos()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:537 +0x287
  github.com/pingcap/tidb/executor.newSlowQueryRetriever()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:54 +0x213
  github.com/pingcap/tidb/executor.parseSlowLog()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:67 +0x49
  github.com/pingcap/tidb/executor.(*testExecSuite).TestParseSlowLogFile()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:166 +0x7ea
  runtime.call16()
      /snap/go/7954/src/runtime/asm_amd64.s:550 +0x3d
  reflect.Value.Call()
      /snap/go/7954/src/reflect/value.go:337 +0xd8
  github.com/pingcap/check.(*suiteRunner).forkTest.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:850 +0xb3b
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:739 +0x11d

Previous write at 0x00c000901b00 by goroutine 96:
  [failed to restore the stack]

Goroutine 228 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).forkTest()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:832 +0x1c4
  github.com/pingcap/check.(*suiteRunner).doRun()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:666 +0x144
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:650 +0x105

Goroutine 96 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).runFunc()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:751 +0x84
  github.com/pingcap/check.(*suiteRunner).runFixture()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:795 +0x3d
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:647 +0x86
==================
==================
WARNING: DATA RACE
Read at 0x00c001057380 by goroutine 228:
  github.com/pingcap/tidb/infoschema.(*Builder).createSchemaTablesForDB()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:559 +0x25e
  github.com/pingcap/tidb/infoschema.(*Builder).InitWithDBInfos()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:537 +0x287
  github.com/pingcap/tidb/executor.newSlowQueryRetriever()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:54 +0x213
  github.com/pingcap/tidb/executor.parseSlowLog()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:67 +0x49
  github.com/pingcap/tidb/executor.(*testExecSuite).TestParseSlowLogFile()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:166 +0x7ea
  runtime.call16()
      /snap/go/7954/src/runtime/asm_amd64.s:550 +0x3d
  reflect.Value.Call()
      /snap/go/7954/src/reflect/value.go:337 +0xd8
  github.com/pingcap/check.(*suiteRunner).forkTest.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:850 +0xb3b
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:739 +0x11d

Previous write at 0x00c001057380 by goroutine 96:
  [failed to restore the stack]

Goroutine 228 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).forkTest()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:832 +0x1c4
  github.com/pingcap/check.(*suiteRunner).doRun()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:666 +0x144
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:650 +0x105

Goroutine 96 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).runFunc()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:751 +0x84
  github.com/pingcap/check.(*suiteRunner).runFixture()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:795 +0x3d
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:647 +0x86
==================
==================
WARNING: DATA RACE
Read at 0x00c000010ea0 by goroutine 228:
  github.com/pingcap/tidb/infoschema/perfschema.initTableIndices()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/perfschema/tables.go:209 +0xca
  github.com/pingcap/tidb/infoschema/perfschema.createPerfSchemaTable()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/perfschema/tables.go:150 +0x405
  github.com/pingcap/tidb/infoschema/perfschema.tableFromMeta()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/perfschema/tables.go:134 +0x199
  github.com/pingcap/tidb/infoschema.(*Builder).createSchemaTablesForDB()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:562 +0x32c
  github.com/pingcap/tidb/infoschema.(*Builder).InitWithDBInfos()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:537 +0x287
  github.com/pingcap/tidb/executor.newSlowQueryRetriever()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:54 +0x213
  github.com/pingcap/tidb/executor.parseSlowLog()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:67 +0x49
  github.com/pingcap/tidb/executor.(*testExecSuite).TestParseSlowLogFile()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:166 +0x7ea
  runtime.call16()
      /snap/go/7954/src/runtime/asm_amd64.s:550 +0x3d
  reflect.Value.Call()
      /snap/go/7954/src/reflect/value.go:337 +0xd8
  github.com/pingcap/check.(*suiteRunner).forkTest.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:850 +0xb3b
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:739 +0x11d

Previous write at 0x00c000010ea0 by goroutine 96:
  [failed to restore the stack]

Goroutine 228 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).forkTest()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:832 +0x1c4
  github.com/pingcap/check.(*suiteRunner).doRun()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:666 +0x144
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:650 +0x105

Goroutine 96 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).runFunc()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:751 +0x84
  github.com/pingcap/check.(*suiteRunner).runFixture()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:795 +0x3d
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:647 +0x86
==================
==================
WARNING: DATA RACE
Read at 0x00c0003fe3c0 by goroutine 228:
  github.com/pingcap/tidb/infoschema/perfschema.initTableIndices()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/perfschema/tables.go:210 +0xef
  github.com/pingcap/tidb/infoschema/perfschema.createPerfSchemaTable()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/perfschema/tables.go:150 +0x405
  github.com/pingcap/tidb/infoschema/perfschema.tableFromMeta()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/perfschema/tables.go:134 +0x199
  github.com/pingcap/tidb/infoschema.(*Builder).createSchemaTablesForDB()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:562 +0x32c
  github.com/pingcap/tidb/infoschema.(*Builder).InitWithDBInfos()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:537 +0x287
  github.com/pingcap/tidb/executor.newSlowQueryRetriever()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:54 +0x213
  github.com/pingcap/tidb/executor.parseSlowLog()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:67 +0x49
  github.com/pingcap/tidb/executor.(*testExecSuite).TestParseSlowLogFile()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:166 +0x7ea
  runtime.call16()
      /snap/go/7954/src/runtime/asm_amd64.s:550 +0x3d
  reflect.Value.Call()
      /snap/go/7954/src/reflect/value.go:337 +0xd8
  github.com/pingcap/check.(*suiteRunner).forkTest.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:850 +0xb3b
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:739 +0x11d

Previous write at 0x00c0003fe3c0 by goroutine 96:
  [failed to restore the stack]

Goroutine 228 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).forkTest()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:832 +0x1c4
  github.com/pingcap/check.(*suiteRunner).doRun()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:666 +0x144
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:650 +0x105

Goroutine 96 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).runFunc()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:751 +0x84
  github.com/pingcap/check.(*suiteRunner).runFixture()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:795 +0x3d
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:647 +0x86
==================
==================
WARNING: DATA RACE
Read at 0x00c0006bf800 by goroutine 228:
  github.com/pingcap/tidb/infoschema/perfschema.initTableIndices()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/perfschema/tables.go:213 +0x12d
  github.com/pingcap/tidb/infoschema/perfschema.createPerfSchemaTable()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/perfschema/tables.go:150 +0x405
  github.com/pingcap/tidb/infoschema/perfschema.tableFromMeta()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/perfschema/tables.go:134 +0x199
  github.com/pingcap/tidb/infoschema.(*Builder).createSchemaTablesForDB()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:562 +0x32c
  github.com/pingcap/tidb/infoschema.(*Builder).InitWithDBInfos()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:537 +0x287
  github.com/pingcap/tidb/executor.newSlowQueryRetriever()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:54 +0x213
  github.com/pingcap/tidb/executor.parseSlowLog()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:67 +0x49
  github.com/pingcap/tidb/executor.(*testExecSuite).TestParseSlowLogFile()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:166 +0x7ea
  runtime.call16()
      /snap/go/7954/src/runtime/asm_amd64.s:550 +0x3d
  reflect.Value.Call()
      /snap/go/7954/src/reflect/value.go:337 +0xd8
  github.com/pingcap/check.(*suiteRunner).forkTest.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:850 +0xb3b
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:739 +0x11d

Previous write at 0x00c0006bf800 by goroutine 96:
  [failed to restore the stack]

Goroutine 228 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).forkTest()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:832 +0x1c4
  github.com/pingcap/check.(*suiteRunner).doRun()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:666 +0x144
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:650 +0x105

Goroutine 96 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).runFunc()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:751 +0x84
  github.com/pingcap/check.(*suiteRunner).runFixture()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:795 +0x3d
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:647 +0x86
==================
==================
WARNING: DATA RACE
Read at 0x00c0003fe3e3 by goroutine 228:
  github.com/pingcap/tidb/table/tables.NewIndex()
      /home/genius/project/src/github.com/pingcap/tidb/table/tables/index.go:111 +0x48
  github.com/pingcap/tidb/infoschema/perfschema.initTableIndices()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/perfschema/tables.go:213 +0x155
  github.com/pingcap/tidb/infoschema/perfschema.createPerfSchemaTable()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/perfschema/tables.go:150 +0x405
  github.com/pingcap/tidb/infoschema/perfschema.tableFromMeta()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/perfschema/tables.go:134 +0x199
  github.com/pingcap/tidb/infoschema.(*Builder).createSchemaTablesForDB()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:562 +0x32c
  github.com/pingcap/tidb/infoschema.(*Builder).InitWithDBInfos()
      /home/genius/project/src/github.com/pingcap/tidb/infoschema/builder.go:537 +0x287
  github.com/pingcap/tidb/executor.newSlowQueryRetriever()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:54 +0x213
  github.com/pingcap/tidb/executor.parseSlowLog()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:67 +0x49
  github.com/pingcap/tidb/executor.(*testExecSuite).TestParseSlowLogFile()
      /home/genius/project/src/github.com/pingcap/tidb/executor/slow_query_test.go:166 +0x7ea
  runtime.call16()
      /snap/go/7954/src/runtime/asm_amd64.s:550 +0x3d
  reflect.Value.Call()
      /snap/go/7954/src/reflect/value.go:337 +0xd8
  github.com/pingcap/check.(*suiteRunner).forkTest.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:850 +0xb3b
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:739 +0x11d

Previous write at 0x00c0003fe3e0 by goroutine 96:
  [failed to restore the stack]

Goroutine 228 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).forkTest()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:832 +0x1c4
  github.com/pingcap/check.(*suiteRunner).doRun()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:666 +0x144
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:650 +0x105

Goroutine 96 (running) created at:
  github.com/pingcap/check.(*suiteRunner).forkCall()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:734 +0x4cc
  github.com/pingcap/check.(*suiteRunner).runFunc()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:751 +0x84
  github.com/pingcap/check.(*suiteRunner).runFixture()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:795 +0x3d
  github.com/pingcap/check.(*suiteRunner).asyncRun.func1()
      /home/genius/go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:647 +0x86
==================
```